src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -92,7 +92,7 @@ export class Kernel implements IKernel {
         return this._info;
     }
     get status(): KernelMessage.Status {
-        return this.notebook?.session?.status ?? 'unknown';
+        return this.notebook?.session?.status ?? (this.isKernelDead ? 'dead' : 'unknown');
```

Comment:
If the IJupyterSession is dead, then Kernel.status will not reflect the status correctly.
Previously it would return `unknown` and that's incorrect. In jupyter `unknown` is actually `not started` and they have `dead` for dead kernels.